include/inviwo/core/util/raiiutils.h

```diff
@@ -39,11 +39,18 @@ namespace util {
 
 class IVW_CORE_API KeepTrueWhileInScope {
 public:
-    KeepTrueWhileInScope(bool* b) : variable_(b) { if(variable_) (*variable_) = true; }
-    ~KeepTrueWhileInScope() { if(variable_) (*variable_) = false; }
```

Comment:
Are we sure we don’t depend on this behavior anywhere?

Reply:
Cannot guarantee that. People might have a different interpretation of what `KeepTrueWhileInScope` should be doing (see comment by @r-englund in #204)

Reply:
Still wondering though why the canvas was shown in the first place. Maybe something within Qt changed slightly and the visibility changes now trigger signals. Who knows ...

Reply:
I would be very careful with this, might break serialisation in hard to notice ways...

Reply:
And it might be incorrect in many others as well.